### PR TITLE
Feature/upgrade sentry cx 4428

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem
We are using an older version of Sentry tracker. Sentry recommends upgrading to @sentry/browser

# Solution
Upgraded to sentry/browser making sure that all functionalities are still working

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [ ] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [ ] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by joninvski** [comment]
Shouldn't     "raven-js": "^3.26.3" dependency be removed?

**Comment by stefaniacardenas** [comment]
Yes, it should. I forgot!

File: src/Tracker/index.js [hunk 1]
```diff
@@ -10,26 +10,6 @@ let shouldSendEvents = false
...
-  breadcrumbCallback: (crumb) => {
-    const isOnfidoXhr = crumb.category === 'xhr' && isOnfidoHostname(crumb.data.url)
-
-    const isOnfidoClick = crumb.category === 'ui.click' && crumb.message.includes('.onfido-sdk-ui')
-
-    const shouldReturnCrumb = isOnfidoXhr || isOnfidoClick
-
-    return shouldReturnCrumb ? crumb : false
-  },
-  whitelistUrls: [/onfido[A-z.]*\.min.js/g],
```
Comment (joninvski) [inline src/Tracker/index.js]: small note, in the view sentry version you also have this setting:

https://docs.sentry.io/error-reporting/configuration/?platform=browser#whitelist-urls

But i'm ok with only adding it in next story as chatted before

File: src/Tracker/index.js [hunk 2]
```diff
@@ -47,18 +27,38 @@ const setUp = () => {
...
+  if (sentryClient) {
+    sentryClient.close(2000)
+  }
   woopra.dispose()
   shouldSendEvents = false
 }
 
 const install = () => {
-  RavenTracker.install()
+  Sentry.init({
```
Comment (joninvski) [inline src/Tracker/index.js]: Sorry, i was just thinking, can this init of sentry mess up clients that also use sentry to track their own web apps?

File: src/Tracker/index.js (hunk 1 above)
Comment (stefaniacardenas) [reply src/Tracker/index.js]: I can bring it back but I don't think it's working

Comment (stefaniacardenas) [reply src/Tracker/index.js]: I'll put it back just in case, then in the next story I can assess if it's work correctly or it needs to be removed

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://869-pr-onfido-sdk-ui-onfido.surge.sh

File: src/Tracker/index.js (hunk 2 above)
Comment (joninvski) [reply src/Tracker/index.js]: @stefaniacardenas any idea about this? Did it already did that?

Comment (stefaniacardenas) [reply src/Tracker/index.js]: I think you've got a point there. I will make sure our Sentry integration does not clash with our client's.

File: src/Tracker/index.js [hunk 3]
```diff
@@ -47,18 +27,38 @@ const setUp = () => {
...
     {sdk_version} : {sdk_version, client})
-
-  Raven.TraceKit.collectWindowErrors = true//TODO scope exceptions to sdk code only
 }
 
 const uninstall = () => {
-  RavenTracker.uninstall()
+  const sentryClient = Sentry.getCurrentHub().getClient();
+  if (sentryClient) {
+    sentryClient.close(2000)
```
Comment (kopijunkie) [inline src/Tracker/index.js]: Is there a reason for a 2 second wait (that's my assumption about what's happening here)?

File: src/Tracker/index.js (hunk 2 above)
Comment (kopijunkie) [reply src/Tracker/index.js]: out of curiosity how would we ensure this? 🤔

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

File: src/Tracker/index.js (hunk 2 above)
Comment (stefaniacardenas) [reply src/Tracker/index.js]: I created multiple instances of Sentry and I noticed that the second instance was overriding the first one. Luckily, Sentry has a way of fixing this by using `BrowserClient`.

File: src/Tracker/index.js (hunk 3 above)
Comment (stefaniacardenas) [reply src/Tracker/index.js]: Not really. It's the maximum time you want to wait for the clients to wrap up sending the last events

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://869-pr-onfido-sdk-ui-onfido.surge.sh

File: src/Tracker/index.js (hunk 3 above)
Comment (kopijunkie) [reply src/Tracker/index.js]: Ah, gotcha.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-869-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://869-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**